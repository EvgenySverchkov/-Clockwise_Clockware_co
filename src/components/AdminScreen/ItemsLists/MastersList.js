import React from "react";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";

import {townsInit} from "../../../store/adminPanel/towns/actions";
import {initMasters, deleteMasterFromState} from "../../../store/adminPanel/masters/actions"
import deleteDataFromServer from "../services/deleteDataFromServer";
import List from "../List";

import { SERVERDOMAIN } from "../../../services/serverUrls";

const MastersList = ({history}) => {
  const state = useSelector(state=>{
    return {mastersArr: state.master_reducer.masters}
  });
  const dispatch = useDispatch();

  function getMastersAndTownsFromServerToState() {
    const headers = {
      Authorization: "Bearer " + sessionStorage.getItem("token") || "",
      "Content-Type": "application/json",
    };
    fetch(`${SERVERDOMAIN}/masters`, { headers })
      .then((data) => data.json())
      .then((data) => dispatch(initMasters(data)))
      .catch((err) => {
        alert("Please, authorise");
        dispatch(initMasters([]));
      });
    getTownsFromServerToState();
  }
  function getTownsFromServerToState() {
    const headers = {
      Authorization: "Bearer " + sessionStorage.getItem("token") || "",
      "Content-Type": "application/json",
    };
    fetch(`${SERVERDOMAIN}/towns`, { headers })
      .then((json) => json.json())
      .then((data) => dispatch(townsInit(data)));
  }
  function deleteMasterById(masterId) {
    deleteDataFromServer(`${SERVERDOMAIN}/masters/delete/${masterId}`)
      .then((data) => {
        if (data.success) {
          dispatch(deleteMasterFromState(data.payload));
          alert(data.msg);
        } else {
          alert(data.msg);
        }
      })
      .catch((err) => alert(err));
  }
  return (
    <List
      dataArr={state.mastersArr}
      deleteAction={deleteMasterById}
      mainRows={["name", "rating"]}
      getData={getMastersAndTownsFromServerToState}
      history={history}
    />
  );
};

MastersList.propTypes = {
  history: PropTypes.object
};

export default MastersList;
