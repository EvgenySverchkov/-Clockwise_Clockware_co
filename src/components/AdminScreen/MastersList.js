import React from 'react';
import {connect} from "react-redux";

import List from "./List";
import ListItem from "./ListItem";

import {deleteMaster} from "../../store/adminPanel/actions";
import withOptions from "../../hocs/withOptions";

import {serverDomain} from "../../services/serverUrls";

function MastersList(props){
  function deleteMasterById(masterId){
    fetch(`${serverDomain}/delete_master/${masterId}`, {
      method: "delete"
    }).then(data=>data.json())
    .then(data=>props.deleteMaster(data))
  }
  ListItem.displayName = "MasterItem";
  let ListItemWithOptions = withOptions(ListItem, deleteMasterById);
  return <List ListItem = {ListItemWithOptions}
               dataArr={props.mastersArr}
               style = {style}/>
}
let style = {
  width: '70%',
  margin: '0 auto',
  tableLayout: 'fixed'
}

export default connect(null, {deleteMaster})(MastersList);
