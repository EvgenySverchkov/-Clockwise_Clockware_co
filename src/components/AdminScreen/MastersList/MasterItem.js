import React from 'react';

export default ({masterObj})=>{
  return (
    <>
      <th scope="row">{masterObj.id}</th>
      <td>{masterObj.name}</td>
      <td>{masterObj.rating}</td>
      <td style={{wordWrap: 'break-word'}}>{masterObj.towns}</td>
    </>
  );
}
