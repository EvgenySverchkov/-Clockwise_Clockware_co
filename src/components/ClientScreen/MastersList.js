import React from 'react';
import {connect} from "react-redux";
import {Link} from 'react-router-dom';

function MastersList(props){
  if(props.mastersArr.length === 0){
    return (
      <>
        <div className="text-left display-4">We haven't masters in this city</div>
        <Link to="/client" className="btn btn-primary mt-5">Сome back</Link>
      </>
    );
  }
  return (
    <form onSubmit={props.submitHandler}>
      <div className="row">
      {
        props.mastersArr.map(item=>{
          return (
            <div className="col-sm-4" key={item.id}>
              <div className="card">
                <div className="card-body">
                  <h5 className="card-title">{item.name}</h5>
                  <p className="card-text">Rating: {item.rating}</p>
                  <p className="card-text">Towns: {item.towns}</p>
                  <div className="form-check form-check-inline">
                    <input className="form-check-input" type="radio" name="chooseMaster" id={item.id} value={item.id}/>
                    <label className="form-check-label" htmlFor={item.id}>Choose</label>
                  </div>
                </div>
              </div>
            </div>
          );
        })
      }
      </div>
    <div className="form-group float-left mt-5 ml-5">
        <Link to="/client" className="btn btn-primary">Сome back</Link>
    </div>
    <div className="form-group float-right mt-5 mr-5">
        <input type="submit" value="Book now" className="btn btn-primary"/>
    </div>
    </form>
  );
}

function mapStateToProps(state){
  return {
    mastersArr: state.client_order_reduser.suitableMasters,
    currentOrder: state.client_order_reduser.currentOrder
  }
};

export default connect(mapStateToProps)(MastersList);
