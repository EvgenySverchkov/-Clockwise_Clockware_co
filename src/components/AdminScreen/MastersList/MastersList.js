import React from 'react';
import MasterItem from "./MasterItem";
import {connect} from "react-redux";
import {deleteMaster} from "../../../store/adminPanel/actions";
import withDeleteBtn from "../../../hocs/withDeleteBtn";

function MastersList(props){
  let MasterItemWithDelete = withDeleteBtn(MasterItem, props.deleteMaster);
  return (
    <ul style={{listStyle: "none"}}>{
      props.mastersArr.map((item)=><MasterItemWithDelete key={item.id+1} masterObj = {item} by={item.id}/>)
    }</ul>
  );
}

export default connect(null, {deleteMaster})(MastersList);
