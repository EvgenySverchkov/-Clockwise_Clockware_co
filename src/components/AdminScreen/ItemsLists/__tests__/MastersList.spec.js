import React from "react";
import {create, act} from "react-test-renderer";
import {Provider} from "react-redux";
import mockStoreConfig from "redux-mock-store";
import {MemoryRouter} from "react-router-dom";
   
import MastersList from "../MastersList";

const mockStore = mockStoreConfig();

describe("Test of <MastersList />", ()=>{
    global.fetch = jest.fn(()=>Promise.resolve({json: jest.fn(()=>Promise.resolve())}));
    const initialState = {
        masterReducer: {
            masters: [{
                id: 1,
                name: "Name",
                rating: 5
            }]
        },
        adminModalWindows: {
            listIsLoad: false
        }
    }
    const mockProps = {
        history: {
            location: "/"
        }
    }
    it("When list is already loaded", async()=>{
        const store = mockStore(initialState);
        let component;
        await act(async ()=>{
            component = await create(
                <Provider store={store}>
                    <MemoryRouter initialEntries={["/"]}>
                        <MastersList {...mockProps}/>
                    </MemoryRouter>
                </Provider>
            );
        })
        expect(component.toJSON()).toMatchSnapshot();
    });
    it("When list is loading", async()=>{
        const store = mockStore({masterReducer:{masters: []}, adminModalWindows: {listIsLoad: true}});
        let component;
        await act(async ()=>{
            component = await create(
                <Provider store={store}>
                    <MemoryRouter initialEntries={["/"]}>
                        <MastersList {...mockProps}/>
                    </MemoryRouter>
                </Provider>
            );
        })
        expect(component.toJSON()).toMatchSnapshot();
    });
});