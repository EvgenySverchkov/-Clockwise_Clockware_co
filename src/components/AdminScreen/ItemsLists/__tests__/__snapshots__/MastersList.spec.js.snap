// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <MastersList /> When list is already loaded 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Rating
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          Name
        </td>
        <td
          style={
            Object {
              "wordWrap": "break-word",
            }
          }
        >
          5
        </td>
        <td
          className="text-center"
        >
          <div
            className="dropdown"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              className="btn btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              id="dropdownMenuButton"
              type="button"
            >
              Options
            </button>
            <div
              aria-labelledby="dropdownMenuLink"
              className="dropdown-menu"
            >
              <button
                className="dropdown-item"
                onClick={[Function]}
              >
                Delete
              </button>
              <a
                className="dropdown-item"
                href="/1"
                onClick={[Function]}
              >
                Edit
              </a>
              <button
                className="dropdown-item"
                data-target="#fullInfoModal"
                data-toggle="modal"
                onClick={[Function]}
                type="button"
              >
                Show full info
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test of <MastersList /> When list is loading 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table table-dark"
  >
    <thead>
      <tr>
        <th>
          Loading...
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
